### PR TITLE
Bridger2

### PR DESCRIPTION
**Pull request by ylv-io**
# Description

### Summary

This PR introduces support for the SuperBridge for the `depositBySig`, `depositERC20`, and `depositETH` functions. These functions will immediately bridge the deposit to the user's wallet on the Kinto chain.

### Implementation Details

- **Contract Renaming and Imports:**
  - Renamed `KintoMainnetMigration2DeployScript` to `DeployBridgerScript`.
  - Added new imports such as `Constants` from `@kinto-core-script/migrations/mainnet/const.sol`.

- **Contract Enhancements:**
  - Implemented `Constants` interface in `DeployBridgerScript` and `UpgradeBridgerScript`.
  - Removed redundant address fetching and added new constructor parameters for `Bridger` contract.
  - Introduced new constants for various addresses like `L2_VAULT`, `EXCHANGE_PROXY`, `WETH`, `DAI`, `USDE`, `SUSDE`, and `WSTETH`.

- **Functionality Additions:**
  - Added `whitelistFinalAssets` function to `Bridger` contract to manage final allowed assets.
  - Enhanced `depositBySig` and `depositETH` functions to include new parameters and logic for bridging assets.
  - Introduced `_swap` and `_stakeUSDe` private functions for handling asset swaps and staking.

- **Script Updates:**
  - Deleted `03-enable_swaps.s.sol` script as it is no longer needed.
  - Updated `04-upgrade_bridger.s.sol` and `05-whitelist_assets.s.sol` scripts to align with new contract changes.
  - Added new script `const.sol` for defining constant addresses.

- **Test Enhancements:**
  - Refactored existing tests to align with new contract changes.
  - Added new tests for `depositERC20`, `depositETH`, and `whitelistFinalAssets` functions.
  - Introduced `BridgeMock` contract for testing bridging functionality.
  - Updated `BridgerHarness` to include new constructor parameters.

- **Code Cleanup and Refactoring:**
  - Removed deprecated and redundant code.
  - Improved code readability and maintainability by refactoring functions and adding comments.
  - Ensured all tests pass with the new changes and added necessary assertions.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [x] Breaking change 
- [ ] Dependency changes
- [ ] Deployment
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: src/bridger/Bridger.sol [hunk 1]
```diff
@@ -241,81 +257,103 @@ contract Bridger is
...
+        uint256 amount,
+        uint256 minReceive,
+        bytes calldata swapCallData
     ) private returns (uint256 amountBought) {
-        amountBought = _amount;
-        if (_finalAsset == sUSDe) _finalAsset = USDe; // if sUSDE, swap to USDe & then stake
-        if (_finalAsset != _inputAsset) {
-            if (!swapsEnabled) revert SwapsDisabled();
+        amountBought = amount;
+        if (inputAsset != finalAsset) {
```
Comment (fedealconada) [inline src/bridger/Bridger.sol:317]: Shouldn't this be `if (inputAsset == finalAsset)`?

File: src/bridger/Bridger.sol [hunk 2]
```diff
@@ -241,81 +257,103 @@ contract Bridger is
...
+        }
+
+        if (inputAsset == ETH && finalAsset == wstETH) {
+            return _stakeEthToWstEth(amount);
+        }
+
+        if (inputAsset == ETH) {
+            WETH.deposit{value: amount}();
+            inputAsset = address(WETH);
+        }
```
Comment (fedealconada) [inline src/bridger/Bridger.sol:328]:
```suggestion
        if (inputAsset == ETH) {
            if (finalAsset == wstETH) {
                return _stakeEthToWstEth(amount);
            }
          WETH.deposit{value: amount}();
          inputAsset = address(WETH);
        }
```

Comment (fedealconada) [reply src/bridger/Bridger.sol:328]: thoughts?

File: src/bridger/Bridger.sol (hunk 1 above)
Comment (fedealconada) [inline src/bridger/Bridger.sol:317]:
```suggestion
        if (inputAsset == finalAsset) {
```

Comment (ylv-io) [reply src/bridger/Bridger.sol:317]: That is why need a fork tests for the swap. Lmao. I was wondering how these tests work.

**Review by fedealconada — APPROVED** [review]
lgtm!

**Review by rrecuero — APPROVED** [review]

**Pull request merged**